homeassistant/components/zha/api.py

```diff
@@ -244,6 +273,103 @@ def async_load_api(hass, application_controller, zha_gateway):
         SCHEMA_WS_RECONFIGURE_NODE
     )
 
+    @websocket_api.async_response
+    async def websocket_get_bindable_devices(hass, connection, msg):
```

Comment:
Would be a lot nicer if we could define these outside of this function. Can't you get the gateway object from `hass.data` ?

Reply:
I can make that work if it is preferable

Reply:
Preference:
 - Easier to read the code. Instead of reading it as F1…F2…F1…F3…F1 it's F1…F2…F3
 - Able to test smaller pieces of code

Reply:
Makes perfect sense. Thanks for the explanation!

Reply:
can I do this and the new registration style in another PR to minimize the scope? I'll do it right after...

Reply:
how about I do this here: https://github.com/home-assistant/home-assistant/pull/21533 instead